lib/node_modules/@stdlib/repl/lib/multiline_handler.js

```diff
@@ -155,6 +162,95 @@ setNonEnumerableReadOnly( MultilineHandler.prototype, '_moveCursor', function mo
 	this._rli.cursor = x;
 });
 
+/**
+* Inserts previous command matching the prefix from history.
+*
+* @private
+* @name _prevCommand
+* @memberof MultilineHandler.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( MultilineHandler.prototype, '_prevCommand', function prevCommand() {
+	var cmd;
+	var i;
+
+	// If we are starting from zero, save the prefix for this cycle...
+	if ( this._history.index === 0 ) {
+		this._history.prefix = this._rli.line.slice( 0, this._rli.cursor );
+	}
+	// Traverse the history till we find the command with a common prefix...
+	while ( this._history.index < this._history.list.length / 3 ) {
+		cmd = this._history.list[ this._history.list.length - ( 3 * this._history.index ) - 2 ]; // eslint-disable-line max-len
+		if ( startsWith( cmd, this._history.prefix ) ) {
+			this.clearInput();
+
+			// For each newline, trigger a `return` keypress in paste-mode:
+			cmd = cmd.split( '\n' );
+			this._multiline.pasteMode = true;
+			for ( i = 0; i < cmd.length - 1; i++ ) {
+				this._rli.write( cmd[ i ] );
+				this._rli.write( null, {
+					'name': 'return'
+				});
+			}
+			this._rli.write( cmd[ cmd.length - 1 ] );
+			this._multiline.pasteMode = false;
```

Comment:
Looks like this code gets duplicated in `nextCommand`. Perhaps it should be extracted to a common method which then gets called from within the loops?